x-pack/plugins/cases/server/services/cases/transform.ts

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/* eslint-disable @typescript-eslint/naming-convention */
+
+import {
+  SavedObject,
+  SavedObjectReference,
+  SavedObjectsBulkResponse,
+  SavedObjectsBulkUpdateResponse,
+  SavedObjectsFindResponse,
+  SavedObjectsUpdateResponse,
+} from 'kibana/server';
+import { ACTION_SAVED_OBJECT_TYPE } from '../../../../actions/server';
+import { ESCaseAttributes, ExternalServicesWithoutConnectorId } from '.';
+import {
+  CaseSavedObjectReference,
+  connectorIdReferenceName,
+  pushConnectorIdReferenceName,
+} from '..';
+import { CaseAttributes, CaseFullExternalService, noneConnectorId } from '../../../common';
+import {
+  findConnectorIdReference,
+  transformFieldsToESModel,
+  transformESConnectorOrUseDefault,
+  transformESConnectorToExternalModel,
+} from '../transform';
+
+export function transformUpdateResponsesToExternalModels(
+  response: SavedObjectsBulkUpdateResponse<ESCaseAttributes>
+): SavedObjectsBulkUpdateResponse<CaseAttributes> {
+  return {
+    ...response,
+    saved_objects: response.saved_objects.map((so) => ({
+      ...so,
+      ...transformUpdateResponseToExternalModel(so),
+    })),
+  };
+}
+
+export function transformUpdateResponseToExternalModel(
```

Comment:
What is an external model?

Reply:
Yeah I really struggled with how to name these functions. I was thinking `ExternalModel` would be the representation of an object outside of the service layer. Basically an object that has already been converted into the form for external (outside of the service layer) to interact with it (so it should have the connector.fields as an object instead of key/value pairs, and have the ids populated on the object).

I was trying to avoid using the actual name of io-ts type just in case we changed them in the future (e.g. ESConnector, CaseAttributes etc).

Reply:
Make sense! Thanks!